designs/plugin-resolution.md

```diff
@@ -0,0 +1,88 @@
+# Plugin and project version resolution
+
+## Old behavior
+
+The same process is followed for all commands, without distinction between `init` and the rest. The only difference is
+that, for `init`, the configuration file will not be present.
+
+1. Obtain the project version from project configuration file's `version` field, `--project-verison` flag, or default
+   project version configured during CLI creation (`cli.New(cli.WithDefaultProjectVersion(...))`).
+    * In case both a project configuration file and the flag are found, fail if they are different.
+2. Obtain the plugin keys from the project configuration file's `layout` field, `--plugins` flag, or default plugins
+   configured during CLI creation for the above project version (`cli.New(cli.WithDefaultPlugins(...))`).
+    * In case both a project configuration file and the flag are found, fail if they are different.
```

Comment:
So, are you saying that if we use a plugin to scaffold the project and that was not found for the CLI it should fail? 

if I scaffold the project with SDK and I used the layout plugins:go/3,pluginA,pluginB, for example, IMO I should still able to use KB. The projects are both compatible. So, could we not only executed the plugins found and raise a warning to let the user knows that in this example the pluginA and B were not found.

Reply:
This is a description of the current behavior, not the proposed enhancement.

And what I'm saying is that, if right now, you call `kubebuilder init --plugins go/v3 ...` and then you run `kubebuilder create api --plugins X ...` where X is anything different than `go/v3`, it fails. This is how it works right now, the proposal suggests not to do this.

Reply:
Thank you for the clarification.